### ISSUE TITLE
TernaryBERT Paper Reproduce Issue (CoLA Task)

### ISSUE THREAD

**Issue by MarsJacobs** [huawei-noah/Pretrained-Language-Model#150]
Hi, Thanks for this great work and answering my previous questions.

I reproduced TernaryBERT on GLUE task, but one of the task result I reproduced is below your result .
This table is the result I got using TernaryBERT git code.

- I used TernaryBERT Git Repository Code (no modification).
- For RTE, CoLA, MRPC Task Results, I used Data Augmentation. (Augmented Data is made from TinyBERT code)
[https://github.com/huawei-noah/Pretrained-Language-Model/blob/master/TinyBERT/data_augmentation.py](url)
- For task specific fine tuning BERT model file for initialization, I modified quant_task_glue.py as follows. (remove student model part, only use GT Label Loss)
           
            #student_logits, student_atts, student_reps = student_model(input_ids, segment_ids, input_mask) 
            teacher_logits, teacher_atts, teacher_reps = teacher_model(input_ids, segment_ids, input_mask)
            lprobs = torch.nn.functional.log_softmax(teacher_logits, dim=-1)
            loss = torch.nn.functional.nll_loss(lprobs, label_ids, reduction='sum')

-  As you can see, Every TernaryBERT task(except CoLA)'s GLUE score is almost same as your paper score.
- But there is still some differences between your CoLA mcc score(Difference is 2.4) and my CoLA mcc score(Difference is 4.3).

<google-sheets-html-origin style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">

### TernaryBERT Paper Results and Reproduce Results Table

Task | BERT base Paper | TernaryBERT Paper | Diff | BERT base | TernaryBERT | Diff
-- | -- | -- | -- | -- | -- | --
**CoLA** | 58.1 | 55.7 | 2.4 | 61.31 | **57.01** | **4.3**
RTE | 71.1 | 72.9 | -1.8 | 73.28 | 73.28 | 0
SST-2 | 93.1 | 92.8 | 0.3 | 93.58 | 92.08 | 1.5
MNLI(F1/acc) | 84.5/84.9 | 83.3/83.3 | 0.8 | 84.125 | 83.91 | 0.22
MRPC(F1/acc) | 90.6/86.5 | 91.2/87.5 | -0.8 | 87.32 | 86.85 | 0.47
QQP(F1/acc) | 87.5/90.9 | 86.7/90.1 | 0.8 | 89.54 | 88.99 | 0.55

</google-sheets-html-origin>

### Data Augmentation Effect Table
<google-sheets-html-origin style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">

Task | FP | w/o Aug | w/ Aug | Aug Effect | Diff with FP
-- | -- | -- | -- | -- | --
CoLA | 61.31 | 47.59 | 57.01 | 9.42 | 4.3
RTE | 73.28 | 70.03 | 73.28 | 3.25 | 0
MRPC | 87.32 | 86.62 | 86.85 | 0.23 | 0.47

</google-sheets-html-origin>

Is there any additional settings regarding TernaryBERT CoLA task fine tuning?
It would be much better if I can get better CoLA task result as TernaryBERT paper. (2.4 mcc difference with FP BERT)

I use default CoLA fine tuning settings(batch size, max sequence length) in [https://github.com/huawei-noah/Pretrained-Language-Model/blob/823ca18efd38bca7a5bcb80a197b11583165dbd3/TernaryBERT/quant_task_glue.py#L197](url)  
default learning rate (2e-05), 1 Epoch (Data Augmentation Setting)

Any information about CoLA fine-tuning setting would be very helpful! 
Thank you.
